adam/perception/visual_perception.py

```diff
@@ -165,25 +212,52 @@ def from_mapping(
                             weight=strokes_map["confidence_score"],
                         )
                     )
+            relative_properties: Dict[str, List[GraphNode]] = {}
+            if cluster_map["relative_distance"] is not None:
+                for other_object_name, relative_distance in cluster_map[
+                    "relative_distance"
+                ].items():
+                    other_object_id = object_name_to_id(other_object_name)
+                    relative_properties[other_object_id] = []
```

Comment:
Minor: I would like to work on the readability here. In the current flow we have to assume in the `relative_size` code that every object has been assigned a list value in the dict already. This makes sense if I think about it, but I'd rather not have to.

One solution might be to separate out this initialization step. Another might be to use `dict.setdefault()` in both loops.

Reply:
@spigo900 I fixed it by switching `relative_properties` to a `defaultdict(list)`. Hope that works; I'm gonna try to merge today

Reply:
`defaultdict` also works. Probably a good idea to convert it back when storing it in the `ClusterPerception` so that typos downstream of this code don't get hidden.

Reply:
Looks good to me. Thank you.